### ISSUE TITLE
Identical simple names and type names

### ISSUE THREAD

**Issue by marek-safar** [dotnet/roslyn#19939]
**Version Used**: 2.3.0.61719 (5fbdd5c5)

**Steps to Reproduce**:

```csharp
using System;
using static TestClass;

internal class Program
{
    public static void Main (string[] args)
    {
        Directions.Up (); // Fails even if there is using static and X.Y member has to be invocable
        TestClass.Directions.Up (); // Works
    }
}

public enum Directions
{
}

public static class TestClass
{
    public static int Directions;
}

public static class Extensions
{
    public static string Up (this int arg)
    {
        return "";
    }
}
```

**Expected Behavior**:

Same behaviour as in other cases where identical simple name and type-name resolution resolution is required.

**Actual Behavior**:

error CS0117: 'Directions' does not contain a definition for 'Up'

Inconsistent behaviour

**Comment by alrz** [comment]
The binder won't go back in a member access if it fails. `Directions` is bound to the type in scope (due to the precedence over `using static`), and then lookup for `Up` fails which is reported.

**Comment by jaredpar** [comment]
This behavior is "By Design". Not sure if we have a public version of the spec containing the using static changes but here is a draft form of the rules that come into play here:

- Otherwise, if the namespaces and type declarations imported by the using-namespace-directives and using-static-directives of the namespace declaration contain exactly one accessible type or non-extension-method static member having name I and K type parameters, then the simple-name refers to that type or member constructed with the given type arguments.

- Otherwise, if the namespaces and types imported by the using-namespace-directives of the namespace declaration contain more than one accessible type or non-extension-method static member having name I and K type parameters, then the simple-name is ambiguous and an error occurs.

Essentially there is a ambiguity here for `Direction` between the `using` and `using static` rules. Hence the rule considering `using static` fails and we fall to the next rule which only includes `using`. At that point `Direction` is unambiguous and refers to the `Enum`.
